rpcserver.go

```diff
@@ -2728,11 +2727,68 @@ func (r *rpcServer) GetNetworkInfo(ctx context.Context,
 		minChannelSize = 0
 	}
 
-	// TODO(roasbeef): graph diameter
+	// Calculate graph diameter: Diameter = maximum length of shortest path
```

Comment:
I know the other network stats are currently computed in-line, but I think this should be lifted to be a method of `ChannelRouter` itself. This way we can cover it with a set of unit tests!

Reply:
@Roasbeef : I've done all the changes, feel free to review again.